### PR TITLE
Authenticate getaccounts

### PR DESCRIPTION
**Pull request by rachedbenmustapha**
This uses a new POST route for GetAccounts in vault, that authenticates and authorizes the call and works with session tokens.

The new auth is opt-in, existing code will not use it unless it is changed to call `.enableIAMOnAdminRoutes()`

### CODE REVIEW

File: bin/vaultclient [hunk 1]
```diff
@@ -190,6 +195,21 @@ program
...
     }));
 
+program
+    .command('get-accounts')
+    .option('--account-ids <ACCOUNT ID,ACCOUNT ID,...>')
+    .option('--canonical-ids <CANONICAL ID,CANONICAL ID,...>')
+    .option('--emails <EMAIL,EMAIL,...>')
+    .action(action.bind(null, 'get-accounts', (client, args) => {
+        client.enableIAMOnAdminRoutes().getAccounts(
+            args.accountIds ? args.accountIds.split(',') : null,
```
Comment (miniscruff) [inline bin/vaultclient]: Given we are in node 16 now, I think you can do optional chaining here.
```suggestion
            args.accountIds?.split(',') || null,
```

**Review by miniscruff — COMMENTED** [review]
Just one comment related to a node 16 feature

File: bin/vaultclient (hunk 1 above)
Comment (rachedbenmustapha) [reply bin/vaultclient]: Sweet!

**Review by miniscruff — APPROVED** [review]

**Review by naren-rajendran — APPROVED** [review]

**Pull request merged**